src/Controller/OchaKeyFiguresController.php

```diff
@@ -233,128 +233,9 @@ public function query(string $provider, string $path, array $query = [], bool $u
-              $data[$figure['figure_id']]['value'] = ($data[$figure['figure_id']]['value'] + $figure['value']) / 2;
-              break;
-
-            case 'list':
-              // Value is comnma separated list.
-              $values = explode(',', $data[$figure['figure_id']]['value']);
-              $values = array_map('trim', $values);
-
-              $new_values = explode(',', $figure['value']);
-              $new_values = array_map('trim', $new_values);
-              $data[$figure['figure_id']]['value'] = implode(', ', array_unique(array_merge($values, $new_values)));
-              break;
-
-            default:
-              // @todo needs more logic.
-              $data[$figure['figure_id']]['value'] += $figure['value'];
-
-          }
-          $data[$figure['figure_id']]['figure_list'][] = $figure;
-          $data[$figure['figure_id']]['cache_tags'] += $this->getCacheTags($figure);
-          $data[$figure['figure_id']]['cache_tags'] = array_unique($data[$figure['figure_id']]['cache_tags']);
-        }
-      }
-
-    }
-
-    return $data;
-  }
-
-  /**
-   * Get aggregated figure by figure id.
-   */
-  public function getFigureByFigureId(string $provider, array $iso3, string $year, string $figure_id) : array {
-    // Fetch data.
-    $prefix = $this->getPrefix($provider);
-
-    $query = [
-      'iso3' => $iso3,
-      'year' => $year,
-      'figure_id' => $figure_id,
-    ];
-
-    $figures = $this->getData($prefix, $query);
-
-    $data = [];
-    if (!empty($figures)) {
-      foreach ($figures as $figure) {
-        if (empty($data)) {
-          $data = $figure;
-          $data['figure_list'] = [];
-          $data['cache_tags'] = $this->getCacheTags($figure);
-        }
-        else {
-          switch ($data['value_type']) {
-            case 'amount':
-            case 'numeric':
-              $data['value'] += $figure['value'];
-              break;
-
-            case 'percentage':
-              $data['value'] = ($data['value'] + $figure['value']) / 2;
-              break;
-
-            case 'list':
-              // Value is comnma separated list.
-              $values = explode(',', $data['value']);
-              $values = array_map('trim', $values);
-
-              $new_values = explode(',', $figure['value']);
-              $new_values = array_map('trim', $new_values);
-              $data['value'] = implode(', ', array_unique(array_merge($values, $new_values)));
-              break;
-
-            default:
-              // @todo needs more logic.
-              $data['value'] += $figure['value'];
-
-          }
-          $data['figure_list'][] = $figure;
-          $data['cache_tags'] += $this->getCacheTags($figure);
-        }
-      }
-
-      $data['cache_tags'] = array_unique($data['cache_tags']);
-    }
+  public function getFigure(string $provider, string $id) : array {
+    $data = $this->query($provider, $id);
+    $data['cache_tags'] = $this->getCacheTags($data);
```

Comment:
```suggestion
    $data['cache_tags'] = !empty($data) ? $this->getCacheTags($data) : [];
```

If the data from the `query()` call is empty, `getCacheTags()` generates array key missing errors.

Maybe the check could be added in `getCacheTags()` instead of here.